File: backend/src/main/java/com/backend/domain/payment/repository/PaymentRepository.java
package com.backend.domain.payment.repository;

import com.backend.domain.payment.entity.Payment;
import com.backend.domain.payment.entity.PaymentStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface PaymentRepository extends JpaRepository<Payment, Long> {

    Optional<Payment> findByOrderId(String orderId);
    Optional<Payment> findByPaymentKey(String paymentKey);

    List<Payment> findAllByOrderByApprovedAtDesc();
    long countByStatus(PaymentStatus status);
}

File: backend/src/main/java/com/backend/domain/payment/repository/PaymentRepository.kt
package com.backend.domain.payment.repository

import com.backend.domain.payment.entity.Payment
import com.backend.domain.payment.entity.PaymentStatus
import org.springframework.data.jpa.repository.JpaRepository

interface PaymentRepository : JpaRepository<Payment, Long> {
    fun findByOrderId(orderId: String): Payment?
    fun findByPaymentKey(paymentKey: String): Payment?

    fun findAllByOrderByApprovedAtDesc(): List<Payment>
    fun countByStatus(status: PaymentStatus): Long
}
